pytorch_lightning/trainer/data_loading.py

```diff
@@ -441,9 +440,7 @@ def request_dataloader(
     def _add_sampler_metadata_collate(dataloader: DataLoader) -> None:
         """Wrap default collate function to retrive ``FastForwardSampler`` state dict when fault tolerant is
         enabled."""
-        dataloader.collate_fn = partial(
-            _capture_metadata_collate, dataset=dataloader.dataset, default_collate=dataloader.collate_fn
-        )
+        _add_capture_metadata_collate(dataloader)
```

Comment:
If `_add_sampler_metadata_collate` only calls `_add_capture_metadata_collate`, shouldn't all occurrences of the former be replaced with the latter?

Reply:
Yes, entirely right !